aspnet-core/src/AbpCompanyName.AbpProjectName.Core/AppVersionHelper.cs

```diff
@@ -21,7 +21,16 @@ public class AppVersionHelper
         /// </summary>
         public static DateTime ReleaseDate
         {
-            get { return new FileInfo(typeof(AppVersionHelper).GetAssembly().Location).LastWriteTime; }
+            get
+            {
+                if (_cachedReleaseDate.HasValue)
+                    return _cachedReleaseDate.Value;
+
+                _cachedReleaseDate = new FileInfo(typeof(AppVersionHelper).GetAssembly().Location).LastWriteTime;
+
+                return _cachedReleaseDate.Value;
+            }
         }
+        private static DateTime? _cachedReleaseDate;
```

Comment:
hi @iyilm4z 
How about using `Lazy`?

```c#
public class AppVersionHelper
{
	/// <summary>
	/// Gets current version of the application.
	/// It's also shown in the web page.
	/// </summary>
	public const string Version = "4.7.1.0";

	/// <summary>
	/// Gets release (last build) date of the application.
	/// It's shown in the web page.
	/// </summary>
	public static DateTime ReleaseDate => LzyReleaseDate.Value;

	private static readonly Lazy<DateTime> LzyReleaseDate = new Lazy<DateTime>(() => new FileInfo(typeof(AppVersionHelper).GetAssembly().Location).LastWriteTime);
}
```

Reply:
@maliming thanks for your review, it makes sense, i wish you recommended me this snippet instead of giving me thump up [here](https://github.com/aspnetboilerplate/aspnetboilerplate/issues/4587#issuecomment-586129221)😄

Reply:
😅 Sorry, I thought that was just a simple idea. 

When you implement it, you may need to consider the beauty of the code. 👍

Reply:
Sometimes i like to implement things explicitly, this was one of them😅